### PR TITLE
Fix/network management flow fixes

### PR DESCRIPTION
**Pull request by svojsu**
Added routings:

- Show manage custom network bottom sheet;
- Route to edit network;
- Show success alert on successful network add;
- Route back to networks list after successful network add;

Features:

- Skip remote nodes for networks if they were previously added as custom;
- Filter added networks in known networks list;
- Modify previously added network if user wants to add already existing custom network with different parameters;
- Switch to auto-balanced mode if user removes current selected custom node;

UI fixes:

- `chainUrlPlaceholder` changed to just a scheme;
- banner inner content spacings changed according to design;
- animations fixed to keep running after screen appear back in stack;
- Add Network button fix inactive state on appear back in stack;

### CODE REVIEW

File: novawallet/Modules/NetworkManagement/KnownNetworksList/KnownNetworksListViewController.swift [hunk 1]
```diff
@@ -33,6 +33,11 @@ final class KnownNetworksListViewController: UIViewController, ViewHolder {
         super.viewDidLoad()
 
         setup()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
         presenter.setup()
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/KnownNetworksList/KnownNetworksListViewController.swift:41]: I would rename ```presenter.setup``` to ```presenter.becameActive``` to explicitly say that it might be called several times during lifecycle

File: novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsInteractor.swift [hunk 2]
```diff
@@ -314,27 +307,65 @@ private extension NetworkDetailsInteractor {
...
+
+        if node == currentSelectedNode {
+            let chainWithAutoBalance = chain.updatingConnectionMode(for: .autoBalanced)
+
+            let setAutoBalanceOperation = repository.saveOperation(
+                { [chainWithAutoBalance] },
+                { [] }
+            )
+
+            let deleteNodeOperation = repository.saveOperation(
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsInteractor.swift:349]: Didn't get why do we need two operations, why just not have a single operation that saves the updated chain?

File: novawallet/Modules/NetworkManagement/NetworksList/NetworksListViewController.swift [hunk 3]
```diff
@@ -104,21 +111,29 @@ extension NetworksListViewController: UITableViewDataSource {
             )
         }
     }
+
+    func tableView(_: UITableView, viewForFooterInSection _: Int) -> UIView? {
+        UIView()
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/NetworksList/NetworksListViewController.swift:116]: What we are trying to fix with that approach?

File: novawallet/Common/Services/ChainRegistry/ChainModelConversion.swift [hunk 4]
```diff
@@ -34,12 +34,26 @@ final class ChainModelConverter: ChainModelConversionProtocol {
...
+
+            customNodes.append(node)
+            customNodesUrlSet.insert(node.url)
+        }
+
         let remoteNodes = remoteModel.nodes
-            .enumerated()
-            .map { ChainNodeModel(remoteModel: $1, order: Int16($0)) }
-        
+            .compactMap { node -> ChainNodeModel? in
```
[the remark this answers is not in the document]
Comment (svojsu) [reply novawallet/Common/Services/ChainRegistry/ChainModelConversion.swift:49]: wanted to make it in single run instead of 2, but I agree, readability if much better with `.map` + `.filter`

File: novawallet/Modules/NetworkManagement/CustomNetwork/Interactor/CustomNetworkBaseInteractor.swift [hunk 5]
```diff
@@ -39,15 +39,60 @@ class CustomNetworkBaseInteractor: NetworkNodeCreatorTrait,
...
+
+    func modify(
+        _ existingNetwork: ChainModel,
+        node: ChainNodeModel,
+        url: String,
+        name: String,
+        currencySymbol: String,
+        chainId: String?,
+        blockExplorerURL: String?,
+        coingeckoURL _: String?
```
[the remark this answers is not in the document]
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/CustomNetwork/Interactor/CustomNetworkBaseInteractor.swift:55]: There is changes with coingecko url parsing and priceId extracting in the [following PR](https://github.com/novasamatech/nova-wallet-ios/pull/1126/commits/20fe5a6ffbdee04711ed595fd62c5a3988f626c1)

File: novawallet/Modules/NetworkManagement/CustomNetwork/Interactor/CustomNetworkBaseInteractor.swift [hunk 6]
```diff
@@ -235,63 +280,66 @@ private extension CustomNetworkBaseInteractor {
...
-                
+
         return String(coingeckoUrlTemplate[matchedRange])
     }
-    
+
     func createExplorer(from url: String?) -> ChainModel.Explorer? {
-        guard let url = url?.trimmingCharacters(in: CharacterSet(charactersIn: "/")) else {
+        guard let url = url?.trimmingCharacters(
+            in: CharacterSet(charactersIn: "/").union(CharacterSet.whitespaces)
```
[the remark this answers is not in the document]
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/CustomNetwork/Interactor/CustomNetworkBaseInteractor.swift:310]: `.slash` is a case of the `String.Separator` enum so it might be a bit misleading to use `String.Separator.slash.rawValue`. Also, we will have to wrap it with `String`, cause `.slash` has `Character` type and `CharacterSet` init needs `String` value. I find the result being too verbose in that case

File: novawallet/Modules/NetworkManagement/NetworksList/NetworksListViewController.swift (hunk 3 above)
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/NetworksList/NetworksListViewController.swift:116]: It's just to create a space between sections according to the design, so we set a footer of the desired height for that.

File: novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsInteractor.swift (hunk 2 above)
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsInteractor.swift:349]: There is some flaky behavior with single operation approach. Sometimes we don't receive `.connected` event on `WebSocketEngineDelegate` when updating connectionMode with removing node at the same time. 

Fixed, but further investigation and tests required.

File: novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsInteractor.swift [hunk 7]
```diff
@@ -314,27 +307,65 @@ private extension NetworkDetailsInteractor {
...
+        _ node: ChainNodeModel,
+        chain: ChainModel
+    ) -> CompoundOperationWrapper<Void> {
+        let operation: (
+            targetOperation: BaseOperation<Void>,
+            dependencies: [BaseOperation<Void>]
+        )
+
+        if node == currentSelectedNode {
+//            let setAutoBalanceOperation = repository.saveOperation(
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsInteractor.swift:342]: Please, remove comments once you fix the issue

File: novawallet/Modules/NetworkManagement/KnownNetworksList/KnownNetworksListViewFactory.swift [hunk 8]
```diff
@@ -2,29 +2,32 @@ import Foundation
 import SoraFoundation
 
 struct KnownNetworksListViewFactory {
-    static func createView() -> KnownNetworksListViewProtocol? {
+    static func createView(
+        successAddPresenting: (wireframe: ModalAlertPresenting, view: ControllerBackedProtocol)
```
Comment (ERussel) [inline novawallet/Modules/NetworkManagement/KnownNetworksList/KnownNetworksListViewFactory.swift:6]: I thought we will improve the logic in all places not only in the CustomNetwork module - don't see a need to move ModalPresenting stuff around. Probably, let's do it in a separate PR.

**Review by ERussel — APPROVED** [review]

File: novawallet/Modules/NetworkManagement/KnownNetworksList/KnownNetworksListViewFactory.swift (hunk 8 above)
Comment (svojsu) [reply novawallet/Modules/NetworkManagement/KnownNetworksList/KnownNetworksListViewFactory.swift:6]: Damn, I was sure I removed it everywhere else. Thanks, I will remove it in separate PR for sure.
